### ISSUE TITLE
Style/ConditionalAssignment cop crash saying undefined method `begin_type?' for :e:Symbol

### ISSUE THREAD

**Issue by milouse** [bbatsov/rubocop#3801]
## Expected behavior

I'm using rubocop with the following configuration (excerpt) : 

```yaml
Style/ConditionalAssignment:
  EnforcedStyle: assign_inside_condition
```

I check a rails controller containing this kind of code (excerpt) : 

```ruby
if params[:finished]
  if params[:finished] == 'yes'
    projects = projects.finished
  else
    projects = projects.unfinished
  end
end
```
Rubocop shouldn't detect any error here.

## Actual behavior

Rubocop crash with the following stack trace (obtained with `rubocop -d`)

```
An error occurred while Style/ConditionalAssignment cop was inspecting /home/milouse/Applis/[snip]/[snap]/app/controllers/api/projects_controller.rb.
undefined method `begin_type?' for :e:Symbol
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/style/conditional_assignment.rb:289:in `assignment_node'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/style/conditional_assignment.rb:236:in `check_assignment_to_condition'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/style/conditional_assignment.rb:222:in `block (2 levels) in <class:ConditionalAssignment>'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:41:in `block (2 levels) in on_lvasgn'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:96:in `with_cop_error_handling'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:40:in `block in on_lvasgn'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:39:in `each'
/home/milouse/.rvm/gems/ruby-2.1.2/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:39:in `on_lvasgn'
[…]
```

## Steps to reproduce the problem

See Expected behavior

## RuboCop version

0.46.0 (using Parser 2.3.2.0, running on ruby 2.1.2 x86_64-linux)

**Comment by pocke** [comment]
Hi. thanks for your feedback!

I can't reproduce the bug.
Maybe error is occurred by other code. Does your code `rescue` expression?

```ruby
begin
  some
rescue => e
end
```

I think the cop crashes while analyse the above code. 
I can reproduce the error with the code.

```
An error occurred while Style/ConditionalAssignment cop was inspecting /tmp/tmp.KyihSJh1i1/test.rb.
undefined method `begin_type?' for :e:Symbol
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/style/conditional_assignment.rb:289:in `assignment_node'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/style/conditional_assignment.rb:236:in `check_assignment_to_condition'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/style/conditional_assignment.rb:222:in `block (2 levels) in <class:ConditionalAssignment>'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:41:in `block (2 levels) in on_lvasgn'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:96:in `with_cop_error_handling'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:40:in `block in on_lvasgn'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:39:in `each'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:39:in `on_lvasgn'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/ast_node/traversal.rb:159:in `block in on_case'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/ast_node/traversal.rb:158:in `each'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/ast_node/traversal.rb:158:in `on_case'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:45:in `on_resbody'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/ast_node/traversal.rb:159:in `block in on_case'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/ast_node/traversal.rb:158:in `each'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/ast_node/traversal.rb:158:in `on_case'
/home/pocke/.gem/ruby/2.3.0/gems/rubocop-0.46.0/lib/rubocop/cop/commissioner.rb:45:in `on_rescue'
```


If your problem is same it, the problem is solved by #3786 .
